server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/camel/CamelProcessor.java

```diff
@@ -90,6 +90,9 @@ public void process(Exchange exchange) throws Exception {
             MailetPipelineLogging.logEndOfMailetProcess(mailet, mail);
             List<MailetProcessorListener> listeners = processor.getListeners();
             long complete = System.currentTimeMillis() - start;
+            if (mail.getRecipients().isEmpty()) {
```

Comment:
If all recipients discard a mail via Sieve, it gets no more match and we end up with a WARNING saying the mail ended up at the end of processor, that it might indicate a misconfiguration.

We don't need to relay emails without recipients. By doing this, we allow mailets to remove recipients.